### PR TITLE
Enchanced Animations

### PR DESCRIPTION
**Pull request by Strflightmight09**

### CODE REVIEW

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 1]
```diff
@@ -21,6 +22,7 @@ public class VisualizeNoteShootingCommand extends Command {
     private Translation2d initialXYVelocity;
     private double initialZVelocity;
     private double noteZ;
+    private boolean crossedFieldWidthBorder = false;
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java]: This is not a right way to name a boolean

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 2]
```diff
@@ -77,6 +83,24 @@ private void configureStartingStats() {
         initialZVelocity = startingPitch.getSin() * startingTangentialVelocity;
     }
 
+    private boolean noteCrossedFieldWithBorder(double noteY) {
+        return noteY > FieldConstants.FIELD_WIDTH_METERS && !crossedFieldWidthBorder;
+    }
+
+    private void configureNoteInAmpStats() {
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java:89]: Insanely cool and a very smart way to go about this

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 3]
```diff
@@ -32,8 +34,12 @@ public void execute() {
         final double timeDifference = Timer.getFPGATimestamp() - startingTimeSeconds;
 
         final Transform3d noteTransform = calculateNoteTransform(timeDifference);
-        final Pose3d notePose = new Pose3d(fieldRelativeNoteExitPointTranslation, new Rotation3d()).plus(noteTransform);
+        Pose3d notePose = new Pose3d(fieldRelativeNoteExitPointTranslation, new Rotation3d()).plus(noteTransform);
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java]: Why not final?

File: src/main/java/frc/trigon/robot/commands/factories/ShootingCommands.java [hunk 4]
```diff
@@ -54,6 +54,12 @@ public static Command getManualLowDeliveryCommand() {
         );
     }
 
+    public static Command getVisualizeNoteShootingCommand() {
+        return new InstantCommand(
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/factories/ShootingCommands.java]: Maybe this should be moved to general commands now?

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 5]
```diff
@@ -77,6 +83,24 @@ private void configureStartingStats() {
         initialZVelocity = startingPitch.getSin() * startingTangentialVelocity;
     }
 
+    private boolean noteCrossedFieldWithBorder(double noteY) {
+        return noteY > FieldConstants.FIELD_WIDTH_METERS && !crossedFieldWidthBorder;
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java]: Width or with? 
I really don't understand this naming. What's the point of `border`?

**Review by levyishai — CHANGES_REQUESTED** [review]

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 6]
```diff
@@ -20,7 +21,8 @@ public class VisualizeNoteShootingCommand extends Command {
     private Rotation2d startingPitch, startingYaw;
     private Translation2d initialXYVelocity;
     private double initialZVelocity;
-    private double noteZ;
+    private Pose3d notePose = new Pose3d();
+    private boolean hasCrossedFieldWidthBorder = false;
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java]: Field width border just doesn't sit right with me, and not very self explanatory. 
Maybe `hasNoteCrossedAmpLine`? 
Anyways, I'd love some more amping videos, so satisfying and cool.

**Review by levyishai — CHANGES_REQUESTED** [review]

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 7]
```diff
@@ -82,12 +82,12 @@ private void configureStartingStats() {
         initialZVelocity = startingPitch.getSin() * startingTangentialVelocity;
     }
 
-    private boolean noteCrossedFieldWidth(double noteY) {
-        return noteY > FieldConstants.FIELD_WIDTH_METERS && !hasCrossedFieldWidthBorder;
+    private boolean noteCrossedCrossedAmpY(double noteY) {
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java]: CrossedCrossed :)
This won't tell us if the note crossed, but if it's the first time it's past the ampy

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 8]
```diff
@@ -22,7 +22,7 @@ public class VisualizeNoteShootingCommand extends Command {
     private Translation2d initialXYVelocity;
     private double initialZVelocity;
     private Pose3d notePose = new Pose3d();
-    private boolean hasCrossedFieldWidthBorder = false;
+    private boolean hasCrossedAmpY = false;
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java:26]: hasNoteCrossedAmpY

**Review by levyishai — CHANGES_REQUESTED** [review]

File: src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java [hunk 9]
```diff
@@ -77,6 +82,24 @@ private void configureStartingStats() {
         initialZVelocity = startingPitch.getSin() * startingTangentialVelocity;
     }
 
+    private boolean noteJustCrossedAmpY(double noteY) {
```
Comment (levyishai) [inline src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java]: didNoteJustCrossAmpY sounds better imo

**Review by levyishai — CHANGES_REQUESTED** [review]

**Review by levyishai — APPROVED** [review]

**Review by Nummun14 — APPROVED** [review]

**Pull request merged**